pkg/generators/snippets/constructor.go

```diff
@@ -24,14 +26,25 @@ func GenerateConstructorForObjectMeta(t *types.Type) (string, generator.Args) {
 	args := generator.Args{
 		"type": t,
 	}
+	argsForPointer(args, tags.IsPointerReceiver(t))
 
 	raw := `// New$.type|raw$ is an autogenerated constructor.
-func New$.type|raw$(name string) *$.type|raw$ {
-	o := &$.type|raw${}
+func New$.type|raw$(name string) $.pointer$$.type|raw$ {
+	o := $.ampersand$$.type|raw${}
 	o.ObjectMeta.Name = name
 	return o
 }
 
 `
 	return raw, args
 }
+
+func argsForPointer(args generator.Args, usePointer bool) {
+	if usePointer {
+		args["pointer"] = "*"
+		args["ampersand"] = "&"
+	} else {
+		args["pointer"] = ""
+		args["ampersand"] = ""
```

Comment:
nit: a function could be created that initializes a set of default args.

Arguments to a function should not be modified.  This function should have a return type of `generator.Args`.

Reply:
Thanks, I replaced this function with `defaultGeneratorArgs` which looks at the input type and returns back the default generator args for it, including the pointer receiver args.
This ends up cleaning up the rest of the setting the args.